src/Aspire.Hosting/OtlpConfigurationExtensions.cs

```diff
@@ -56,8 +58,22 @@ public static void AddOtlpEnvironment(IResource resource, IConfiguration configu
             }
 
             // Set the service name and instance id to the resource name and UID. Values are injected by DCP.
-            context.EnvironmentVariables["OTEL_RESOURCE_ATTRIBUTES"] = "service.instance.id={{- .Name -}}";
-            context.EnvironmentVariables["OTEL_SERVICE_NAME"] = "{{- index .Annotations \"otel-service-name\" -}}";
+            var dcpDependencyCheckService = context.ExecutionContext?.ServiceProvider?.GetService(typeof(IDcpDependencyCheckService)) as IDcpDependencyCheckService;
```

Comment:
Can ExecutionContext and ServiceProvider be null here? Are null forgiving operators needed?

Also, use `GetRequiredService<IDcpDependencyCheckService>()`

Reply:
Thanks, that is much better. Fixed.